src/server/scripts/Spells/spell_item.cpp

```diff
@@ -256,9 +256,15 @@ class spell_item_anger_capacitor : public SpellScriptLoader
                 caster->CastSpell(target, spellId, aurEff);
             }
 
+            void OnRemove(AuraEffect const*, AuraEffectHandleModes)
+            {
+            	GetTarget()->RemoveAurasDueToSpell(SPELL_MOTE_OF_ANGER);
```

Comment:
is target guaranteed on ``AfterEffectRemove`` ?

Reply:
Could you tell me what do you mean with that?

Reply:
I dont know if the return value of GetTarget at that stage of the aura effect lifespan, is guaranteed to be the actual target of the aura or just nullptr. Do you?

Reply:
Isn't [this](https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/game/Spells/Auras/SpellAuras.cpp#L56) enough?

Reply:
nope, unrelated

Reply:
OK, I will add corresponding check.

Reply:
I'm not saying that its needed, I'm saying that I dont know if its needed.